### PR TITLE
Make tftpboot directory dynamic and replace settings via setup.py

### PR DESCRIPTION
**Pull request by watologo1**
https://github.com/cobbler/cobbler/pull/2347

### CODE REVIEW

**Comment by thiller2018** [comment]
Looks good to me

**Comment by watologo1** [comment]
I should have waited with review requests. Check have finally finished, fortunately successfully.

File: cobbler.spec [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Copyright (c) 2006 Michael DeHaan <mdehaan@redhat.com>
```
Comment (SchoolGuy) [inline cobbler.spec:1]: Is there a specific reason you added that line?

Comment (watologo1) [reply cobbler.spec:1]: > Is there a specific reason you added that line?

Please do not squash the patches, then the reason is in a separate submit message:
In openSUSE:Factory there are some additional tests, beside rpmlint. I'll ask whether they can be integrated there, as late error messages are counterproductive. Anyway the bot complains about:
Output of check script:
cobbler.spec does not appear to contain a Copyright comment. Please stick to the format
  
  # Copyright (c) 2011 Stephan Kulow

As Michael DeHaan has submitted the first patch for the spec file and for quite a while was the only submitter, it's certainly him, who should be listed there.

File: setup.py [hunk 2]
```diff
@@ -604,6 +604,17 @@ def run(self):
             # files
             ("%s/grub_config/grub" % libpath, glob("config/grub/grub/*")),
             # dirs
+            ("%s/boot" % tftproot, []),
+            ("%s/etc" % tftproot, []),
+            ("%s/grub" % tftproot, []),
+            ("%s/images" % tftproot, []),
+            ("%s/images2" % tftproot, []),
+            ("%s/ppc" % tftproot, []),
+            ("%s/ppc" % tftproot, []),
```
Comment (hbokh) [inline setup.py:613]: Duplicate line

File: setup.py [hunk 3]
```diff
@@ -604,6 +604,17 @@ def run(self):
...
+            ("%s/s390x" % tftproot, []),
+            ("%s/pxelinux.cfg" % tftproot, []),
+            ("%s/system" % tftproot, []),
```
Comment (hbokh) [inline setup.py:616]: Just to be sure.
So the dirs `system` and `system_link` are moved up from dir. `/var/lib/tftpboot/grub/` into new, separate directories `/var/lib/tftpboot/system` and `/var/lib/tftpboot/system_link`?

Take these files into consideration:
From cobbler-2.2.1-1-3477-g7d45ed59

```
# ag system_link
config/grub/README.grubconfig
51:grub/grub/system_link.*

cobbler/tftpgen.py
280:                        link_path = os.path.join(self.bootloc, "grub", "system_link", system.name)
```

Comment (watologo1) [reply setup.py:616]: > Just to be sure.
> So the dirs `system` and `system_link` are moved up from dir. `/var/lib/tftpboot/grub/` into new, separate directories `/var/lib/tftpboot/system` and `/var/lib/tftpboot/system_link`?

Argh. No, this is a bug/regression. Thanks a lot for the review/heads-up!
I'll send a fix.

**Pull request merged**